File: JavaScript/Day 03/day3-part1/day3-part1.js
const fs = require("fs");
const data = fs
  .readFileSync("input.in", { encoding: "utf8", flag: "r" })
  .toString()
  .split("\r\n")
  .map(String);

let x = 0,
  y = 0,
  countTrees = 0;
const rows = data.length,
  columns = data[0].length;
while (x < rows) {
  if (data[x][y] == "#") {
    countTrees++;
  }
  x++;
  y = (y + 3) % columns;
}

fs.writeFileSync("output.out", countTrees.toString());

File: C++/Day 03/day3-part1/day3-part1.cpp
#include "../../AOCHeaders/stdafx.h"


int main()
{
	std::fstream in("input.in", std::fstream::in);
	std::fstream out("output.out", std::fstream::out);

	std::vector<std::string> map(std::istream_iterator<std::string>{in}, {});
	int x{ 0 }, y{ 0 }, rows{ (int)map.size() }, columns{ (int)map[0].size() }, countTrees{ 0 };

	while (x < rows)
	{
		if (map[x][y] == '#')
		{
			countTrees++;
		}
		x++;
		y = (y + 3) % columns;
	}

	out << countTrees;

	in.close();
	out.close();
}